### PR TITLE
development

### PR DESCRIPTION
**Pull request by hechukwu**
#### What does this PR do

make final PR for project

#### Description of task

create a document management system

#### How can this be tested

N/A

#### Questions

N/A

#### Any background context

N/A

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+DATABASE_URL_DEV=postgres://andeladeveloper:''@127.0.0.1:5432/documents-dev
+DATABASE_URL_TEST=postgres://andeladeveloper:''@127.0.0.1:5432/documents-test
```
Comment (ghost) [inline .env]: Don't push your .env file to git. Always make sure it's the first thing you ignore in your git ignore file

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Place your settings in this file to overwrite default and user settings.
```
Comment (ghost) [inline .vscode/settings.json]: Is this file meant to be in your repo?

File: README.md [hunk 3]
```diff
@@ -1 +1,82 @@
...
+    - Delete any user.
+    - Update any user's records.
+    - Create a new role.
+    - View all created roles.
+    - Delete created roles aside the default roles `admin` and `regular`
+
+
+**Documents**:
+Documents can be created and must have:
+- title
```
Comment (ghost) [inline README.md]: your git readme file needs to show  a list of all the endpoints and their output.

File: client/actions/actionTypes.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+// export const SEARCH_DOCUMENT_REQUEST = 'SEARCH_DOCUMENT_REQUEST';
+// export const SEARCH_DOCUMENT_SUCCESSFUL = 'SEARCH_DOCUMENT_SUCCESSFUL';
+// export const SEARCH_DOCUMENT_FAIL = 'SEARCH_DOCUMENT_FAIL';
+
+// // document
+// export const CREATE_DOCUMENT_REQUEST = 'CREATE_DOCUMENT_REQUEST';
+// export const CREATE_DOCUMENT_SUCCESSFUL = 'CREATE_DOCUMENT_SUCCESSFUL';
+// export const CREATE_DOCUMENT_FAIL = 'CREATE_DOCUMENT_FAIL';
+
```
Comment (ghost) [inline client/actions/actionTypes.js]: you shouldn't push commented code

File: client/actions/authenticationAction.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * logout action creator
+ * @export
+ * @returns {function}
+ */
+export function logout() {
+  return (dispatch) => {
+    localStorage.removeItem('jwtToken');
+    localStorage.removeItem('user');
+    setAuthorizationToken(false);
+    dispatch(setCurrentUser({}));
```
Comment (ghost) [inline client/actions/authenticationAction.js:31]: This is meant to un-set the user in state; A better name could be giving to this function.

File: client/actions/documentActions.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
+import axios from 'axios';
+import * as types from './actionTypes';
+
+/**
+ * Load document success action creator
+ * @export
+ * @param {any} document
```
Comment (ghost) [inline client/actions/documentActions.js]: Is this going to any or an array? Remember The state is immutable. I can see you made an excessive use of "any", try and correct this.

File: client/actions/documentActions.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+/**
+ * Action creator to get all documents accessible to current user
+ * @returns {object} documents
+ */
+export function loadAllDocuments() {
+  return dispatch => axios.get('/documents')
+    .then((response) => {
+      dispatch(loadDocumentSuccess(response.data));
+    }).catch((error) => {
+      throw (error);
```
Comment (ghost) [inline client/actions/documentActions.js]: There should be an error handler that dispatches an action for handling errors

File: client/actions/userActions.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+ * @returns  {object} user
+ */
+export function retrieveUsersSuccess(allUsers) {
+  return {
+    type: types.RETRIEVE_USERS_SUCCESS,
+    allUsers
+  };
+}
+
+export function updateUserSuccess(user) {
```
Comment (ghost) [inline client/actions/userActions.js]: No documentation for this function

File: client/components/AdminComponent/userForm.js [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      errors: {},
+      user: props.user || {},
+    };
+    this.onChange = this.onChange.bind(this);
+    this.updateUser = this.updateUser.bind(this);
+    // this.saveUser = this.saveUser.bind(this);
```
Comment (ghost) [inline client/components/AdminComponent/userForm.js]: Remove comment

**Comment by hechukwu** [comment]
@andela-emurphy Thanks for the comments, I will refactor the codes

**Pull request closed without merging**